hypervisor/arch/x86/guest/vm.c

```diff
@@ -57,16 +57,19 @@ void *get_sworld_memory_base(void)
 	return post_user_vm_sworld_memory;
 }
 
-uint16_t get_vmid_by_uuid(const uint8_t *uuid)
+uint16_t get_vmid_by_name(const char *name)
 {
-	uint16_t vm_id = 0U;
-
-	while (!vm_has_matched_uuid(vm_id, uuid)) {
-		vm_id++;
-		if (vm_id == CONFIG_MAX_VM_NUM) {
-			break;
+	uint16_t vm_id = ACRN_INVALID_VMID;
+	uint16_t matched_cnt = 0;
+
+	/* check if there are duplicate VM names in vm configurations */
```

Comment:
Why do you scan the whole list checking for duplicate names here (every time you lookup a VM by its name to get its ID) instead of doing it once when you assign a VM its ID?

Reply:
here avoid to assign the same VM name, for static configurated VMs and dynamically created VMs, we need to make sure the vm name is unique.